Assets/Main/CameraControls.cs

```diff
@@ -5,48 +5,85 @@
 
 using UnityEngine;
 
-public class CameraControls: MonoBehaviour {
+public class CameraControls : MonoBehaviour {
 	public float perspectiveZoomSpeed = 0.5f;
 	public float orthoZoomSpeed = 0.5f;
 	public Camera cam;
 
-	private Touch touchZero;
-	private Touch touchOne;
+	private Touch t1;
+	private Touch t2;
+
+	private float minMovement;
+	private float currMovement;
 
 	private Vector2 touchZeroPrevPos;
 	private Vector2 touchOnePrevPos;
 
+	private bool firstTouch;
+	private Vector3 worldPin;
+	private Vector3 worldTouch;
+
+
 	private float prevTouchDeltaMag;
 	private float touchDeltaMag;
 
 	private float deltaMagnitudeDiff;
 
+	private MainDirector director;
+
 	public void Start() {
 		cam = gameObject.GetComponent<Camera>();
+		director = GameObject.FindGameObjectWithTag("Director").GetComponent<MainDirector>();
+		minMovement = 1;
+		firstTouch = true;
 	}
 
 	void Update() {
+		if (director == null) {
+			director = GameObject.FindGameObjectWithTag("Director").GetComponent<MainDirector>();
```

Comment:
Should check if MainDirector exists rather than the director gameobject itself. Test to see if this is even needed in the first place.

Reply:
>Should check if MainDirector exists rather than the director gameobject itself. 
wopps I do
>Test to see if this is even needed in the first place.
Works fine without it.